### ISSUE TITLE
useless-suppression false positive for line-too-long when a # appears inside a string literal before the actual pragma comment

### ISSUE THREAD

**Issue by rphakli** [pylint-dev/pylint#11440]
### Bug description

line-too-long's pragma-stripping logic (FormatChecker.check_lines / remove_pylint_option_from_lines, using the OPTION_PO regex) locates the # pylint: disable=... pragma by searching the raw physical line for the first #, then scanning forward for pylint:. If the source line contains a # character inside a string literal before the real trailing pragma comment, OPTION_PO matches starting at that in-string # instead of the real comment, and everything from that point to the end of the line gets treated as "the pragma" and stripped when recomputing the line length.

Reproduction:
`repro.py`:

```python
PATTERN = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx#yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy"  # pylint: disable=line-too-long
```

```
$ pylint --max-line-length=100 --disable=all --enable=line-too-long,useless-suppression repro.py
repro.py:1:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)
```

Removing the `# pylint: disable=line-too-long` comment shows the line genuinely needs it:

```
$ pylint --max-line-length=100 --disable=all --enable=line-too-long,useless-suppression repro.py
repro.py:1:0: C0301: Line too long (133/100) (line-too-long)
```

This looks like the same general bug type as #7911 ("Useful 'line-too-long' suppression considered useless") and the duplicate-marked #10230, both of which show the same disable-makes-it-useless/remove-it-and-its-too-long contradiction, but with an extra trailing *comment* causing the confusion rather than a `#` embedded in string/regex content.

### Configuration

```ini

```

### Command used

```shell
pylint --max-line-length=100 --disable=all --enable=line-too-long,useless-suppression repro.py
```

### Pylint output

```python
repro.py:1:0: I0021: Useless suppression of 'line-too-long' (useless-suppression)
```

### Expected behavior

No errors are expected.

### Pylint version

```shell
pylint 4.0.6
astroid 4.0.4
Python 3.13.7 (main, Oct 14 2025, 11:47:11) [Clang 17.0.0 (clang-1700.0.13.5)]
```

### Additional dependencies

```python

```
